src/classes/RD2_ElevateIntegrationService.cls

```diff
@@ -183,39 +186,226 @@ public inherited sharing class RD2_ElevateIntegrationService {
             SELECT CommitmentId__c
             FROM npe03__Recurring_Donation__c
             WHERE CommitmentId__c IN :oppsByCommitmentId.keySet()]) {
-            
+
             for (Opportunity opp : oppsByCommitmentId.get(rd.CommitmentId__c)) {
                 opp.npe03__Recurring_Donation__c = rd.Id;
             }
         }
     }
 
+
     /**
-    * @description Sends HttpRequest to Elevate to cancel Commitments for close Recurring Donations
+    * @description Cancels commitments in Elevate connected to the closed Recurring Donations
     * @param closedRDs Closed RDs with the user specified closed reason
     * @return void
     */
     public void cancelCommitments(List<npe03__Recurring_Donation__c> closedRDs) {
-        UTIL_Http.RequestService requestService = new UTIL_Http.RequestService();
-        List<UTIL_Http.Response> responses = new List<UTIL_Http.Response>();
+        new CancelCommitmentService().cancelCommitments(closedRDs);
```

Comment:
@RadSFDeveloper How does the UI handle errors? The method is synchronous, but it only writes a message to the log. Is the intent that the page doesn't display a toast or other "error" style message, but instead just reverts the RDStatus to "Active" and updates the StatusReason field with the returned error?

Reply:
This `cancelCommitments` is called from the async `RD2_QueueableService.CancelCommitmentService`. The "Close Recurring Donation" QuickAction is prevented only if there are issues with closing the RD. The "Status Reason" is set to "Pending Elevate Process". When the commitment cancel succeeds, then it is set to the user entered value at the "Close Recurring Donation". On an error, the "Status Reason" is set to the error, but the RD state stays closed.

Reply:
Thanks for the explanation @RadSFDeveloper